Homework 2

EmelyanovPavel opened the pull request:
Homework on Functions of C++ topic

Commit:
Homework from 25-02-2025

tak333 reviewed (changes requested):
Правила именования, правила оформления.
Нет std:: у математических функций, избавьтесь от vector и pair.

Commit:
код без ошибок

EmelyanovPavel committed:
функция void + функция int main с вызовом функции void

EmelyanovPavel committed:
Подключенные библиотеки перенесены под первый комментарий

tak333 commented on Homework on Functions in C++ topic.cpp:
```diff
@@ -0,0 +1,65 @@
+//the distance between which is minimal.
+
+//A function for calculating the distance between two points
+double f(double x1, double y1, double x2, double y2) {
```
Именование должно быть говорящим.

tak333 commented on Homework on Functions in C++ topic.cpp:
```diff
@@ -0,0 +1,65 @@
+    return (a < b) ? a : b;
+}
+
+void task18() {
```
Проверяйте именование функций. Раз задание одно, то просто task или task1.

tak333 commented on Homework on Functions in C++ topic.cpp:
```diff
@@ -0,0 +1,65 @@
+
+    std::cout << "\nMinimal distance: ";
+
+    if (min_distance == d12) {
```
Используйте if ... else ...

EmelyanovPavel committed:
Update Homework on Functions in C++ topic.cpp

Использованы говорящие именования функций double, использована операция if-else и изменено имя функции void.

tak333 commented on Homework on Functions in C++ topic.cpp:
```diff
@@ -2,64 +2,70 @@
 //a function for finding the minimum of two numbers
 double min(double a, double b) {
-    return (a < b) ? a : b;
+    if (a < b) {
```
Тут с тернарным было хорошо, комментарий был к другому.
